daemon/src/model/cfd.rs

```diff
@@ -608,19 +617,9 @@ impl Cfd {
             bail!("Start contract not allowed in version {}", self.version)
         }
 
-        let margin = match self.position {
-            Position::Long => {
-                calculate_long_margin(self.initial_price, self.quantity, self.leverage)
-            }
-            Position::Short => calculate_short_margin(self.initial_price, self.quantity),
-        };
+        let margin = self.margin(self.position);
+        let counterparty_margin = self.margin(self.position.counter_position());
```

Comment:
I think it would be cleaner to create two functions `margin` and `counterparty_margin`.

Reply:
good idea, it certainly improves the ux of this API!